pkg/controller/datavolume-controller.go

```diff
@@ -2779,6 +2787,35 @@ func getDeltaTTL(dv *cdiv1.DataVolume, ttl int32) time.Duration {
 	return delta
 }
 
+// handleCloneWithoutSource does error handling and validation of clones without source PVC
+func (r *DatavolumeReconciler) handleCloneWithoutSource(datavolume *cdiv1.DataVolume) error {
+	sourcePvc, err := r.findSourcePvc(datavolume)
+	if err != nil {
+		// Clone without source
+		if k8serrors.IsNotFound(err) {
+			datavolume.Annotations[AnnNoSourceClone] = "true"
+			r.updateDataVolume(datavolume)
+		}
+		return err
```

Comment:
Why do you return the error and not nil in case of NotFound?

Reply:
If we just return nil there, the clone will fail anyway when trying to get the clone strategy. I need to think about how to handle this, but I see two main options: 
- The current one: We return an error, so the reconcile loop keeps running again and again until the source PVC is created and the clone can be completed. However, I guess we don't want unhandled errors in our reconciler, so I'm not sure about that. Maybe it'll be cleaner to add a `RequeueAfter`?

- The ideal: If the source PVC is not found, we just end the reconcile loop until the source PVC is created, without constantly rerunning the loop or triggering redundant errors. I tried to do this but I don't see an easy way to trigger the reconcile again when the source PVC is created.

Reply:
Thanks for sharing that, I think it's exactly what I wanted. I tried to do it by making the clone own the source PVC, but it was getting very complicated.